openedx/features/pakx/lms/overrides/utils.py

```diff
@@ -446,15 +446,18 @@ def get_course_progress_percentage(request, course_key):
     return _calculate_progress_for_block(get_progress_information(request, course_key))
 
 
-def get_and_save_course_progress(course_key, user_id):
+def get_and_save_course_progress(course_key, user_id=None, request=None):
```

Comment:
When a tab calls for the progress we are sure it doesn't need to be updated, so we need to separate logic for tabs usage.

Reply:
We are calculating new percentage when showing in tab, this was not being saved before